src/core/offscreen-animation-loop.js

```diff
@@ -0,0 +1,167 @@
+/* global window */
+import AnimationLoop, {requestAnimationFrame, cancelAnimationFrame} from './animation-loop';
+import {getPageLoadPromise, createCanvas, getCanvas} from '../webgl-utils';
+
+export default class OffscreenAnimationLoop {
+
+  /*
+   * Create the script for the rendering worker.
+   * @param opts {object} - options to construct an AnimationLoop instance
+   */
+  static createWorker(opts) {
+    return self => {
+
+      self.animationLoop = new AnimationLoop(Object.assign({}, opts, {
+        offScreen: true,
+        // Prevent the animation loop from trying to access DOM properties
```

Comment:
* Maybe this should be enforced via `offScreen: true`.
* In general we should update the `AnimationLoop` to not assume there is a canvas.